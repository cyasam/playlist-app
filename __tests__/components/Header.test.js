import React from 'react'
import { shallow } from 'enzyme'
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { Header, mapStateToProps } from '../../src/scripts/components/Header'

const createMockStore = configureMockStore([thunk])

describe('Header Component', () => {
  const setup = (propOverrides) => {
    const state = {
      authentication: {
        loading: false,
        auth: null,
        error: ''
      }
    }

    const store = createMockStore(state)

    const props = {
      auth: state.auth,
      ...propOverrides
    }

    const wrapper = shallow(<Header {...props} />)
    return {
      wrapper,
      props,
      store,
      state
    }
  }

  it('renders properly', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('has Searchbox Component', () => {
    const { wrapper } = setup()
    expect(wrapper.find('withRouter(Connect(Searchbox))').exists()).toBe(true)
  })

  it('has login button if auth is null', () => {
    const { wrapper } = setup()
    expect(wrapper.find('.login-btn').exists()).toBe(true)
  })

  it('has logout button if auth is not null', () => {
    const { wrapper } = setup({ auth: { user: 'abc' } })
    expect(wrapper.find('.logout-btn').exists()).toBe(true)
  })

  it('returns state from mapStateToProps', () => {
    const { store, state } = setup()
    const result = {
      auth: state.authentication.auth
    }

    expect(mapStateToProps(store.getState())).toEqual(result)
  })
})
