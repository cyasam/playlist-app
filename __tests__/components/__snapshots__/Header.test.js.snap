// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders properly 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-dark bg-primary"
  >
    <Link
      className="navbar-brand"
      replace={false}
      to="/"
    >
      Playlist App
    </Link>
    <Connect(Searchbox) />
    <Link
      className="login-btn btn btn-secondary"
      replace={false}
      to="/login"
    >
      Login
    </Link>
  </nav>
</header>
`;
